Rewrite autoinstallation chapter of client configuration guide (#936)
uyuni-project/uyuni-docs: modules/client-configuration/pages/autoinst-profiles.adoc

For information about {kickstart} profiles, see xref:client-configuration:autoinst-profiles.adoc#kickstart[Kickstart Profiles].

The autoinstallation file contained in the profile can include variables and code snippets.
For information about variables and code snippets, see xref:client-configuration:autoinst-profiles.adoc#templates-syntax[Templates Syntax].

[[declare-profile]]
== Declare the Profile

When you have prepared an autoinstallation file and distribution, you can create profiles to manage autoinstallation on your {productname} Server.
For a given distribution, the profile will determine how to install this distribution.

One way to create a profile is to upload an {ay} or {kickstart} file.
Alternatively, for {kickstart} only, you can use the {webui} wizard.

.Procedure: Creating an Autoinstallation Profile by Upload
. In the {productname} {webui}, navigate to menu:Systems[Autoinstallation > Profiles].
. Click btn:[Upload Kickstart/Autoyast File].
. In the [guimenu]``Label`` field, type a name for the profile.
  Do not use spaces.
. In the [guimenu]``Autoinstall Tree`` field, select the autoinstallation distribution to use for this profile.

> . In the [guimenu]``Virtualization Type`` field, select the type of virtualization to use for this profile, or select ``None`` for no virtualization.

Reviewer: ```suggestion
. In the [guimenu]``Virtualization Type`` field, select the type of virtualization to use for this profile, or select ``None`` if you do not want to use this profile to create a new virtual machine.
```

Reason: Re-install of a virtual machine must use a "normal" profile. The extra values asked are only useful to create a  VM but not for any other reason.

Author: ok.

This can overwrite your customization.
Verify new or changed settings before saving them, by navigating to the [guimenu]``Partitioning`` tab.

The kernel options from the distribution and the profile are combined.

You can change the details and settings of your autoinstallation profiles by navigating to menu:Systems[Autoinstallation > Profiles] and clicking the name of the profile you want to edit.
Alternatively, navigate to menu:Systems[System List], select the client you want to provision, and navigate to the menu:Provisioning[Autoinstallation] subtab.

[[autoyast]]
== AutoYast Profiles

An {ay} profile consists of a [guimenu]``Label`` that identifies the profile, an [guimenu]``Autoinstall Tree`` that points to an autoinstallation distribution, various options, and, most importantly, an {ay} installation file.

The {ay} installation file is an XML file that give directions to the {ay} installer. {ay} calls it a "control file".
For the full syntax of {ay} installation files, see https://doc.opensuse.org/projects/autoyast/#cha-configuration-installation-options.

{suse} provides templates of {ay} installation files that you can use as a starting point for your own custom files.

> You will find them at https://github.com/SUSE/manager-build-profiles.

Reviewer: ```suggestion
You will find them at https://github.com/SUSE/manager-build-profiles in the directoy "AutoYaST".
```

Author: ok

Each of these profiles requires you to set some variables before you use it.
Check the [path]``README`` file included with the script to determine which variables you need.
For more information about using variables in {ay} scripts, see xref:client-configuration:autoinst-profiles#variables[Variables].

> For more information about {ay}, see https://doc.opensuse.org/projects/autoyast/.

Reviewer: I think we should add a sub-section to AutoYaST. It should talk about the important sections in the autoyast file for installing with SUSE Manager. They are:

- `<general>$SNIPPET('spacewalk/sles_no_signature_checks')</general>` 
- `<add-on>` and how to add child channels to the installation
- `<software><products config:type="list"> ...` and how to specify the correct product for the Unified Installer
- The registration snippets for salt and traditional

Author: I just copied this list as a starting point.

I just don't have the time to start writing a full section, but I agree it would be worth it.

Reviewer 2: Created new issue to add this suggestion later: https://github.com/SUSE/spacewalk/issues/14765, so this suggestion does not hold back the approval.

A more recent, Salt-based alternative to AutoYast, is Yomi.
For information about Yomi, see xref:salt:yomi.adoc[Install with Yomi].

[[kickstart]]
== Kickstart Profiles

> {kickstart} profiles offer a larger number of configuration options.

Reviewer: ```suggestion
{kickstart} profiles offer a large number of configuration options.
```

"Larger" sounds like "Kickstart is better than Autoyast as it provide more config options". AFAIK the oposit is true

Author: It's simply a typo - my bad.
